sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -456,10 +456,10 @@ case class MapObjects(
             ($elementJavaType)${genInputData.value}${itemAccessor(loopIndex)};
           $loopNullCheck
 
-          if (${loopVar.isNull}) {
+          ${genFunction.code}
+          if (${genFunction.isNull}) {
```

Comment:
Even `loopVar.isNull` is true, we should still call `genFunction` because it may have special handling for null case, for example `WrapOption` will produce a `None` for null value. We should depend on `genFunction.isNull` to decide `$convertedArray[$loopIndex]` is null or not.